pallets/parachain-staking/src/migrations/v5.rs

```diff
@@ -31,10 +41,30 @@ pub(crate) fn migrate<T: Config>() -> Weight {
 	// Kill selected candidates list
 	old::SelectedCandidates::<T>::kill();
 
-	// count candidates
-	let counter: u32 = CandidatePool::<T>::iter().fold(0, |acc, _| acc.saturating_add(1));
+	// migrate CandidatePool to TopCandidates
+	// DANGER: this needs to happen before we write to the new CandidatePool!
+	let old_candidate_pool = old::CandidatePool::<T>::get();
+	TopCandidates::<T>::put(old_candidate_pool);
+	old::CandidatePool::<T>::kill();
+
+	// Copy over total
+	let total = old::Total::<T>::get();
+	TotalCollatorStake::<T>::set(total);
+	old::Total::<T>::kill();
+
+	// copy & count candidates
+	let counter: u32 = old::CollatorState::<T>::iter().fold(0, |acc, (key, candidate)| {
+		CandidatePool::<T>::insert(&key, candidate);
+		acc.saturating_add(1)
+	});
 	CandidateCount::<T>::put(counter);
 
+	// manually kill the ParachainStaking + CollatorState storage.
+	unhashed::kill_prefix(
```

Comment:
C: Just saying, would it be easier to read also afterwards to add the old types to the deprecated module and call `kill` on them? If they are still in the storage, the kill should still take effect. Plus, by having a type defined we can also make sure that in `pre_upgrade` those values exist and in `post_upgrade` that they don't exist anymore.

Reply:
storagemaps don't have kill. The storage is already in the old `mod`

Reply:
There's a `drain` method, wouldn't that be enough to clear the old storage?

Reply:
I am just saying that we should probably avoid using these hard-coded hex. Even just calling some hex-encoding function on the storage item name would be a better solution.

Reply:
I'm not sure `drain` would also kill the prefix. It only drains the entries.

Reply:
don't know who else that works

Reply:
ah drain would be enough. didn't see that!

Reply:
yeah i also don'T like the hardcoded hex